libeos-payg/provider.c

```diff
@@ -172,7 +173,7 @@ epg_provider_add_code   (EpgProvider  *self,
 
   g_assert (iface->add_code != NULL);
 
-  return iface->add_code (self, code_str, now_secs, error);
+  return iface->add_code (self, code_str, now_secs, time_added, error);
```

Comment:
This function should either `g_return_val_if_fail (time_added != NULL, FALSE)` or it should arrange for a dummy variable to be passed to `iface->add_code` if the parameter is `NULL`. The former is more straightforward. This parameter should also be added to the documentation comment above this function.